aklite-callback-synchronous/start.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/sh
+################################################################
+#
+# Docker handling of aktualizr-lite callbacks
+#
+# Copyright (C) 2021 Foundries.IO
+#
+# SPDX-License-Identifier: MIT
+#
+################################################################
+
+#
+# directory paths for setup and notifying
+#
+CBFILE=${CALLBACK_SCRIPT:-"/var/sota/aklite-callback.sh"}
+TOML=${TOML_FILE:-"/etc/sota/conf.d/z-90-aklite-callback.toml"}
+UPDATES=${UPDATE_DIR:-"/var/lib/aklite-callback"}
+
+#
+# timing values
+# DELAY is the time in seconds between checking for notifications
+#
+DELAY=1
+
+#
+# Debug Logging function: reports date/time to journalctl log
+#
+function debug()
+{
+    if [ -f ${UPDATES}/debug ]; then
+        echo "[DEBUG] $(date +"%b %e %T") aklite-callback(docker): $*"
+    fi
+}
+
+#
+# Logging function: reports date/time to docker log
+#
+function log()
+{
+    echo "$(date +"%b %e %T") aklite-callback(docker): $*"
+}
+
+#
+# Logging function: reports date/time to docker log
+#
+function respond()
+{
+    echo "${1}" > ${2}
+}
+
+#
+# Setup function for activating the callback mechanism
+# with aktualizr-lite
+#
+function setup()
+{
+    #
+    # Make sure to install callback callback file
+    #
+    log "Installing callback"
+    cp `basename ${CBFILE}` ${CBFILE}
+    chmod 755 ${CBFILE}
+    echo -e "[pacman]\ncallback_program = ${CBFILE}" > ${TOML}
+}
+
+###################################################################
+#
+# Start of process
+#
+###################################################################
+
+log "Starting handling of aktualizr-lite callbacks"
+setup
+
+# clear out old messages
+rm -f ${UPDATES}/*.msg
```

Comment:
will this not delete the message the aklite callback script is creating at https://github.com/foundriesio/containers/pull/14/files#diff-89b985d619cf70161aecab96ac425b5d36e16b6bff3d485ac7feeaad044a9496R91

Reply:
This line was to clear out old messages as a way of making sure a "bad" OTA or some other event doesn't leave old messages in the queue when the aklite Docker container starts up.

However, I forget that we will be stopping / starting containers as a part of the update process.  So this will be run mid-update causing problems.

As a matter of fact, I'm going to drop this method of "message sharing" and go back to a Python-based HTTP service.  Most subscribers will be able to use that solution and it presents the nice side effect of: when the app is down, the connection timer can be shorter and recycle quickly to gracefully move forward.

Reply:
Dropped the whole messaging system in favor of http communication which was cleaner